packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -1836,8 +1838,8 @@ class Route extends EmberObject.extend(ActionHandler, Evented) implements IRoute
       // merge in the query params for the route. As a mergedProperty,
       // Route#queryParams is always at least `{}`
 
-      let controllerDefinedQueryParameterConfiguration =
-        (get(controller, 'queryParams') as any) || {};
+      let controllerDefinedQueryParameterConfiguration: ControllerQueryParam[] =
+        (get(controller, 'queryParams') as ControllerQueryParam[]) || [];
```

Comment:
This was incorrectly typed before and the `{}` led to a bug when attempting to use `for of`.